docs/proposal/kmesh_support_localityLB.md

```diff
@@ -0,0 +1,127 @@
+
+What is locality strict mode? In locality strict mode, the LB (load balancing) algorithm will only select backends that exactly match the routingPreference. This means that in such a mode, the load balancer enforces a strict policy where it only routes traffic to backends based on a perfect match with the specified locality preferences, ensuring that requests are handled by servers that meet specific criteria related to their location or other attributes.
+
+### Design Details
+
+Introduce a priority (rank) between the existing service and endpoint to achieve finer-grained traffic control and service management. The newly designed endPointKey will consist of the service ID, priority level, and backend index, with its corresponding value being the backendKey. Development will be carried out in both user space and kernel space to implement the functionality.
+
+![locality_lb_pic](pics/locality_lb.svg)
+
+#### Control plane（user-space）
+1. The user-space kmesh process needs to maintain locality, clusterId, and network information, which is updated when the xDS receives information about the workload currently on that node.
+2. The user-space kmesh process also needs to maintain routingPreference, which is updated when the xDS receives information about the service currently on that node.
+3. Once the above information has been successfully obtained by the user-space, for each incoming workload, its locality information is compared with the kmesh process's locality to calculate a priority.
+   Calculation Process: Match according to the routing rules defined in the service load balancing scope. For example, if a service is configured with a region, the kmesh process's region is compared with the target backend's region. If they are the same, the priority value is incremented. After calculating the rank, the corresponding endpointKey is generated using the service ID, backend ID, and rank, thereby associating the service with the backend.
+
+#### Data plane（kernel-space）
+When a request is made to a service, the BPF program traverses the endpoint map to query the backend ID based on the requested service ID and count, following the priority rank. 
+- In strict mode, only backends that match all scopes are considered.
+- In failover mode, the BPF program will iterate through the ranks starting from the highest and moving downward. If there are backends stored at a particular priority level, one of the queried backends is selected at random to serve the request.
+
+#### data struct
+1. workload.h
+```
+typedef struct {
+    __u32 service_id;    // service id
+    __u32 prio;          // prio means rank, 6 means match all, and 0 means match nothing
```

Comment:
By adding this, how do we select a endpoint now?

Reply:
For random lb mode，workload will only be added to endpoint with maxprio(6).
For locality lb mode，workload will be added to endpoint with rank that calculated by matching kmesh processor's locality info with workload's locality info.
We also record the count number of endpoints belongs to which prio in  serviceValue, so that we can use it as it before. 
In the bpf prog, If service is in random lb mode, we can search endpoint with maxprio. if it is in locality lb mode, we will iter prio from maxprio to 0, if count of that prio >0, which means there have one or more endpoints in that prio, we can choose one workload index by random int value with count, and get endpoint whith serviceId, prio and workloadIndex.

Reply:
make sense, not the bpf map update should be a little bit tricky

Reply:
`0 means nothing` means what?

Reply:
prio value is from 0 to 6, 0 means the lowest priolity.

Reply:
hmm, i would suggest the opposite. Because we can search from the highest priority easily

Reply:
Ok, I have updated it.